dask/array/core.py

```diff
@@ -3325,7 +3325,9 @@ def concatenate(seq, axis=0, allow_unknown_chunksizes=False):
     )
 
     # Drop empty arrays
-    seq = [a for a in seq if a.size]
+    seq2 = [a for a in seq if a.size]
```

Comment:
Why was it necessary to have a new name `seq2` here?

Reply:
The line just afterwards reverts to `seq` if `seq2` is empty.